pack/pack.go

```diff
@@ -46,24 +48,25 @@ type Event struct {
 	Links []httputil.Link `json:"links,omitempty"`
 }
 
-func (p *Pack) generateId() {
+type ok interface {
```

Comment:
Apart from the issue mentioned on the Pack.OK i do dislike this interface rather a lot.